src/shim/ShimUtils.ts

```diff
@@ -111,19 +111,19 @@ export function validateDocumentData(data: Uint8Array) {
 }
 
 /**
- * Validate that the provided encrypted document is in the proper form, which should be Base64 encoded bytes of the document content and the AES IV prepended. Therefore
- * validate the length is at least 1 more than the IV length.
+ * Validate that the provided encrypted document is in the proper form, which should be Base64 encoded bytes or a Uint8Array of the document content and the AES IV prepended.
+ * Therefore validate the length is at least 1 more than the IV length.
  */
-export function validateEncryptedDocument(documentData: Base64Bytes) {
-    if (typeof documentData !== "string") {
-        throw new Error(`Invalid encrypted document format provided. Expected a string.`);
-    }
+export function validateEncryptedDocument(documentData: Base64Bytes | Uint8Array) {
     let encryptedByteLength: number;
+    if (typeof documentData !== "string" && documentData.constructor.name !== "Uint8Array") {
+        throw new Error("Invalid encrypted document content. Content should be a Base64 string or Uint8Array.");
+    }
     try {
-        //This will throw if the documentData isn't valid base64
-        encryptedByteLength = byteLength(documentData);
+        //The byteLength call will throw if the documentData isn't valid base64
+        encryptedByteLength = typeof documentData === "string" ? byteLength(documentData) : documentData.byteLength;
```

Comment:
Seems like doing a single typeof and assigning it to a local variable might make this and line 119 a little more clear.

Reply:
TS doesn't narrow correctly then, because the argument could've been modified between when the check was done and when the call to `byteLength` happens.